crates/cxx-qt-lib/src/core/qvariant/qvariant.cpp

```diff
@@ -24,34 +24,17 @@
 // https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/kernel/qvariant.h?h=v5.15.6-lts-lgpl#n491
 // https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/kernel/qvariant.h?h=v5.15.6-lts-lgpl#n411
 #if (QT_VERSION >= QT_VERSION_CHECK(6, 0, 0))
-
-#if (QT_POINTER_SIZE == 4)
-// 32bit is 3 * 32bit ptr (12) + union with double (8) + 4 bytes padding
-// alignment is 8 byte on 32bit systems as well due to the double
-assert_alignment_and_size(QVariant,
-                          alignof(double),
-                          (sizeof(::std::size_t) * 3) + sizeof(double) +
-                            4 /* compiler padding */);
-#else
-// 64bit is 3 * 64ptr ptr (16) + union with double (8)
-// alignment is 8 bytes from the double or the pointer on 64bit systems
-assert_alignment_and_size(QVariant,
-                          alignof(double),
-                          (sizeof(::std::size_t) * 3) + sizeof(double));
-#endif
-
+constexpr static ::std::array<::std::size_t, 4> arr{ sizeof(::std::size_t),
+                                                     sizeof(::std::size_t),
+                                                     sizeof(::std::size_t),
+                                                     sizeof(double) };
+assert_alignment_and_size(QVariant, alignof(double), arr);
 #else
-
-// 3 * uint (12) + union with double (8)
-// but due to compiler optimisation it ends up as
-// 3 * ushort (6) + union with double (8) + 2 bytes padding
-// alignment is 8 byte on 32bit systems as well due to the double
-assert_alignment_and_size(
-  QVariant,
-  alignof(double),
-  (sizeof(::std::uint16_t /* compiler optimised from ::std::uint32_t */) * 3) +
-    sizeof(double) + 2 /* compiler padding */);
-
+constexpr static ::std::array<::std::size_t, 4> arr{ sizeof(uint),
```

Comment:
as discussed in places like this, i think we should still use `::std::uint16_t` as we are using `u16` in Rust. As we want to check that the Rust size ends up the same as the Qt size. We know that the Qt type uses `uint` so it's not actually checking the right thing if we have `uint` here.

Reply:
as we discussed before, the packed uints should probably be represented as a single u32 rather than three u16's. maybe a task for another PR? or should i go ahead and do it here?

Reply:
nevermind, done.